source/agora/consensus/data/Transaction.d

```diff
@@ -159,6 +136,29 @@ public struct Transaction
     {
         return hashFull(this).opCmp(hashFull(other));
     }
+
+    pure nothrow @nogc:
+
+    /// A `Freeze` transaction is one that has one or more `Freeze` outputs
+    /// If there is more than one output then it is allowed to have a single
+    ///  `Payment` output for a refund of any amount
+    public bool isFreeze () const
```

Comment:
The name is kind of confusing which makes me think that all the outputs are `Freeze`. `hasFreeze`, `containsFreeze` would be better.

Reply:
The transaction is still having the concept of `freeze` though. Although there may be a `Payment` output included for refund, these helpers are used where the logic is at the transaction level rather than the outputs within.

Reply:
Got it. Thanks.